react-app/src/library/services/headset.ts

```diff
@@ -0,0 +1,238 @@
+import { Observable, BehaviorSubject } from 'rxjs';
+import { VendorImplementation, ImplementationConfig } from './vendor-implementations/vendor-implementation';
+import PlantronicsService from './vendor-implementations/plantronics/plantronics';
+import SennheiserService from './vendor-implementations/sennheiser/sennheiser';
+import JabraService from './vendor-implementations/jabra/jabra';
+import JabraNativeService from './vendor-implementations/jabra/jabra-native/jabra-native';
+import ApplicationService from './application';
+import { ConsumedHeadsetEvents } from '../types/consumed-headset-events';
+import { CallInfo } from '../types/call-info';
+import { EventInfo, VendorConversationIdEvent, VendorEvent, HoldEventInfo, MutedEventInfo } from '../types/emitted-headset-events';
+export default class HeadsetService {
+  private static instance: HeadsetService;
+
+  plantronics: VendorImplementation;
+  jabraNative: VendorImplementation;
+  jabra: VendorImplementation;
+  sennheiser: VendorImplementation;
+  application: ApplicationService;
+  selectedImplementation: VendorImplementation;
+  headsetEvents$: Observable<ConsumedHeadsetEvents>;
+
+  private _headsetEvents$: BehaviorSubject<ConsumedHeadsetEvents>;
+  private _implementations: VendorImplementation[] = [];
+  private logger: any;
+
+  private constructor(config: ImplementationConfig) {
+    // super();
+    //TODO: Just a temporary typing for testing purposes
+    this._headsetEvents$ = new BehaviorSubject<ConsumedHeadsetEvents | any>({
```

Comment:
this can probably be cleaned up. I think it should just be `this._headsetEvents$ = new BehaviorSubject<ConsumedHeadsetEvents>(null);`